example/gauss_laguerre_quadrature.cpp

```diff
@@ -26,258 +31,222 @@
+        T j_alpha_m1;
 
-      // Here, we obtain an estimate of the first zero of J_alpha(x).
+        if(alpha < 1.4F)
+        {
+          // For small alpha, use a short series obtained from Mathematica(R).
+          // Series[BesselJZero[v, 1], {v, 0, 3}]
+          // N[%, 12]
+          j_alpha_m1 = (((          T(0.09748661784476F)
+                          * alpha - T(0.17549359276115F))
+                          * alpha + T(1.54288974259931F))
+                          * alpha + T(2.40482555769577F));
+        }
+        else
+        {
+          // For larger alpha, use the first line of Eqs. 10.21.40 in the NIST Handbook.
+          const T alpha_pow_third(boost::math::cbrt(alpha));
+          const T alpha_pow_minus_two_thirds(T(1) / (alpha_pow_third * alpha_pow_third));
+
+          j_alpha_m1 = alpha * (((((                             + T(0.043F)
+                                    * alpha_pow_minus_two_thirds - T(0.0908F))
+                                    * alpha_pow_minus_two_thirds - T(0.00397F))
+                                    * alpha_pow_minus_two_thirds + T(1.033150F))
+                                    * alpha_pow_minus_two_thirds + T(1.8557571F))
+                                    * alpha_pow_minus_two_thirds + T(1.0F));
+        }
 
-      T j_alpha_m1;
+        const T vf             = (  T(order * 4U)
+                                  + T(alpha * 2U)
+                                  + T(2U));
+        const T vf2            = vf * vf;
+        const T j_alpha_m1_sqr = j_alpha_m1 * j_alpha_m1;
 
-      if(alpha < 1.4F)
-      {
-        // For small alpha, use a short series obtained from Mathematica(R).
-        // Series[BesselJZero[v, 1], {v, 0, 3}]
-        // N[%, 12]
-        j_alpha_m1 = (((          0.09748661784476F
-                        * alpha - 0.17549359276115F)
-                        * alpha + 1.54288974259931F)
-                        * alpha + 2.40482555769577F);
-      }
-      else
-      {
-        // For larger alpha, use the first line of Eqs. 10.21.40 in the NIST Handbook.
-        const T alpha_pow_third(boost::math::cbrt(alpha));
-        const T alpha_pow_minus_two_thirds(T(1) / (alpha_pow_third * alpha_pow_third));
-
-        j_alpha_m1 = alpha * (((((                             + 0.043F
-                                  * alpha_pow_minus_two_thirds - 0.0908F)
-                                  * alpha_pow_minus_two_thirds - 0.00397F)
-                                  * alpha_pow_minus_two_thirds + 1.033150F)
-                                  * alpha_pow_minus_two_thirds + 1.8557571F)
-                                  * alpha_pow_minus_two_thirds + 1.0F);
+        first_laguerre_root = (j_alpha_m1_sqr * (   -T(0.6666666666667F)
+                                                 + ((T(0.6666666666667F) * alpha) * alpha)
+                                                 +  (T(0.3333333333333F) * j_alpha_m1_sqr) + vf2)) / (vf2 * vf);
```

Comment:
Doesn't this lose precision? I might do T(2)/T(3), or boost::math::constants::two_thirds.